src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,17 +57,17 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request: Publish => {
         log.debug("publishing %s".format(request))
-        val props = properties getOrElse new AMQP.BasicProperties.Builder().build()
-        sender ! withChannel(channel, request)(c => c.basicPublish(exchange, routingKey, mandatory, immediate, props, body))
+        sender ! withChannel(channel, request)(c => publishAndReturnSeqNo(c, request))
       }
       case request@Transaction(publish) => {
         sender ! withChannel(channel, request) {
           c => {
             c.txSelect()
-            publish.foreach(p => c.basicPublish(p.exchange, p.key, p.mandatory, p.immediate, p.properties getOrElse new AMQP.BasicProperties.Builder().build(), p.body))
+            val seqNos = publish.map(p => publishAndReturnSeqNo(c, p))
```

Comment:
```scala
[ERROR] [04/01/2015 17:32:17.888] [application-akka.actor.default-dispatcher-4] [akka://application/user/$a/$a] shutdown
com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=406, reply-text=PRECONDITION_FAILED - cannot switch from confirm to tx mode, class-id=90, method-id=10)
	at com.rabbitmq.client.impl.ChannelN.asyncShutdown(ChannelN.java:478)
	at com.rabbitmq.client.impl.ChannelN.processAsync(ChannelN.java:315)
	at com.rabbitmq.client.impl.AMQChannel.handleCompleteInboundCommand(AMQChannel.java:144)
	at com.rabbitmq.client.impl.AMQChannel.handleFrame(AMQChannel.java:91)
	at com.rabbitmq.client.impl.AMQConnection$MainLoop.run(AMQConnection.java:550)
	at java.lang.Thread.run(Thread.java:745)```

So the only way it won't fail when you are not in confirm select mode, which way you'll get back a set of zeros